refactor(Merge Weekend create and Confiuration)

lomamech opened the pull request:
- Merge weekend create and configuration modals
- Remove not necessary Option and Test
- Update API in server side
- Updte Integration and E2E test

closes #3783

lomamech committed:
refactor(Merge Weekend create and Confiuration)
- Merge weekend create and configuration modals
- Remove not necessary Option and Test
- Update API in server side
- Updte Integration and E2E test

closes #3783

lomamech committed:
Resolve, clear and sanitaze code

lomamech committed:
Clear code, Use object destructuring prefer-destructuring

jniles commented on client/src/modules/payroll/weekend_configuration/templates/action.tmpl.html:
```diff
@@ -11,12 +11,6 @@
     </li>
 
-    <li>
-      <a data-method="configure" ui-sref="configurationWeekend.config({id : row.entity.id})" href>
```
Can we remove the templates/controllers for this?  It seems like we can now delete whatever files make up the `configurationWeekend.config()` state.

jniles commented on server/config/routes.js:
```diff
@@ -735,7 +735,6 @@ exports.configure = function configure(app) {
   app.post('/weekend_config', weekendConfig.create);
   app.put('/weekend_config/:id', weekendConfig.update);
   app.get('/weekend_config/:id/days', weekendConfig.listConfig);
-  app.post('/weekend_config/:id/days', weekendConfig.createConfig);
```
👍

jniles commented on server/controllers/payroll/weekendConfig/index.js:
```diff
@@ -52,10 +52,22 @@ function detail(req, res, next) {
+
+  delete data.daysChecked;
 
   db.exec(sql, [data])
```
Go ahead and make this function `async/await` so you don't have to use `let`.  Like this:
```js
const { insertId } = await db.exec(sql, [data]);
const dataConfigured = configuration.map(id => ([id, insertId]));
await db.exec('INSERT INTO .... VALUES ?', dataConfigured]);
res.status(201).jsoin({ id : insertId });
```

jniles commented on test/end-to-end/weekendConfig/weekend_config.spec.js:
```diff
@@ -18,14 +18,6 @@ describe('Weekend Configuration Management', () => {
   });
 
-  it('successfully set week days in weekend configuration', async () => {
-    await page.setWeekendConfig(updateWeekendConfigLabel);
```
Is there somewhere that you can remove these functions?  I don't think they will ever be used again.

jniles commented on client/src/js/constants/bhConstants.js:
```diff
@@ -139,13 +139,13 @@ function constantConfig() {
-      { id : 6, label : 'FORM.LABELS.WEEK_DAYS.SATURDAY' },
+      { id : 0, label : 'FORM.LABELS.WEEK_DAYS.SUNDAY', checked : false },
+      { id : 1, label : 'FORM.LABELS.WEEK_DAYS.MONDAY', checked : false },
+      { id : 2, label : 'FORM.LABELS.WEEK_DAYS.TUESDAY', checked : false },
```
👍

jniles reviewed (changes requested):
This is great!  It looks like there may be a lot of legacy code hanging around after this change.  Please remove all functions/controllers/templates that this change will cause to never be used again.

lomamech committed:
refactor(Week end Configuration)
- Remove all functions and routes unused
- Use function async and async/await for create Week end configuration

jniles commented on server/controllers/payroll/weekendConfig/index.js:
```diff
@@ -49,24 +49,42 @@ function detail(req, res, next) {
 
 // POST /WEEKEND_CONFIG
-function create(req, res, next) {
+async function create(req, res) {
```
This is better!  We still need to handle errors though.  To make sure these errors are handled, please add back in "next()' and wrap all your code in a try/catch block.  Like this:
```js
async function create(req, res, next) {
  try {
    // all your code
  } catch (e) {
    next(e);
  }
}
```

jniles commented on test/end-to-end/weekendConfig/weekend_config.page.js:
```diff
@@ -25,28 +32,6 @@ class WeekendConfigPage {
-    await notification.hasSuccess();
-  }
-
-  async setWeekendConfig(label) {
```
👍

jniles commented on test/integration/weekEndConfig.js:
```diff
@@ -77,37 +75,6 @@ describe('(/payroll/weekend_configuration) The /payroll/weekend_configuration  A
 
-  // INTEGRATION TEST FOR SETTING WEEKEND IN CONFIGURATION
-
-  it('POST /WEEKEND__CONFIG/:ID/DAYS should Set WeekEnds in Configuration', () => {
```
👍

jniles reviewed (commented):
Awesome work.  Just one tiny try/catch and then I'll merge it.

lomamech committed:
Handle errors though for create week end configuration
